test/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/CheckCompanyRegistrationControllerSpec.scala

```diff
@@ -0,0 +1,166 @@
+
+    val appId = ApplicationId.random
+    val application = anApplication(id = appId)
+  }
+
+  "checkCompanyRegistrationPage" should {
+    "return 200" in new Setup {
+      val fakeRequest = FakeRequest().withCSRFToken
+
+      val mySubmission = MarkedSubmission(submittableSubmission.submit(), completedProgress, markedAnswers)
+      
+      AuthConnectorMock.Authorise.thenReturn()
+      ApplicationActionServiceMock.Process.thenReturn(application)
+      SubmissionServiceMock.FetchLatestMarkedSubmission.thenReturn(mySubmission)
+
+      val result = controller.page(appId)(fakeRequest)
+      
+      status(result) shouldBe Status.OK
+    }
+
+    "return 400 if the submission is not submitted" in new Setup {
+      val fakeRequest = FakeRequest().withCSRFToken
+
+      val mySubmission = MarkedSubmission(submittableSubmission, completedProgress, markedAnswers)
+
+      AuthConnectorMock.Authorise.thenReturn()
+      ApplicationActionServiceMock.Process.thenReturn(application)
+      SubmissionServiceMock.FetchLatestMarkedSubmission.thenReturn(mySubmission)
+
+      val result = controller.page(appId)(fakeRequest)
+      
+      status(result) shouldBe Status.BAD_REQUEST
+    }
+
+    "return 404 if not found" in new Setup {
+      val fakeRequest = FakeRequest().withCSRFToken
+
+      AuthConnectorMock.Authorise.thenReturn()
+      ApplicationActionServiceMock.Process.thenReturn(application)
+      SubmissionServiceMock.FetchLatestMarkedSubmission.thenNotFound()
+
+      val result = controller.page(appId)(fakeRequest)
+      
+      status(result) shouldBe Status.NOT_FOUND
+    }
+  }
+
+  "checkCompanyRegistrationAction" should {
+    "redirect to correct page when marking URLs as checked" in new Setup {
+      val fakeRequest = FakeRequest()
+                          .withCSRFToken
+                          .withFormUrlEncodedBody("submit-action" -> "checked")
+
+      AuthConnectorMock.Authorise.thenReturn()
+      ApplicationActionServiceMock.Process.thenReturn(application)
+      SubmissionServiceMock.FetchLatestMarkedSubmission.thenReturn(appId)
+      SubmissionReviewServiceMock.UpdateCheckedCompanyRegistrationStatus.thenReturn(SubmissionReview(submissionId, 0))
+
+      val result = controller.action(appId)(fakeRequest)
+
+      status(result) shouldBe SEE_OTHER
+    }
+
+    "redirect to correct page when marking URLs as come-back-later" in new Setup {
+      val fakeRequest = FakeRequest()
+                          .withCSRFToken
+                          .withFormUrlEncodedBody("submit-action" -> "come-back-later")
+
+      AuthConnectorMock.Authorise.thenReturn()
+      ApplicationActionServiceMock.Process.thenReturn(application)
+      SubmissionServiceMock.FetchLatestMarkedSubmission.thenReturn(appId)
+      SubmissionReviewServiceMock.UpdateCheckedCompanyRegistrationStatus.thenReturn(SubmissionReview(submissionId, 0))
+
+      val result = controller.action(appId)(fakeRequest)
+
+      status(result) shouldBe SEE_OTHER
```

Comment:
you can actually check where its getting redirected to by looking at the Location header, like this: https://github.com/hmrc/api-gatekeeper-approvals-frontend/blob/main/test/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/CheckSandboxControllerSpec.scala#L121

Reply:
Good point - added check